src/main/kotlin/lotto/domain/Amount.kt

```diff
@@ -3,7 +3,8 @@ package lotto.domain
 import java.math.BigDecimal
 import java.math.RoundingMode
 
-class Amount(
+@JvmInline
+value class Amount(
     private val value: BigDecimal,
 ) {
```

Comment:
Using value class 👍 
As you may have seen, these special classes are made inline using a wrapper for each inline class. This feature is quite interesting to check out. :)
https://kotlinlang.org/docs/inline-classes.html#representation

Reply:
wow this is actually interesting 👀 Thanks for sharing :)